lib/src/main/java/oiccli/Service.java

```diff
@@ -0,0 +1,443 @@
+        if (httpArgs == null) {
+            httpArgs = new HashMap<>();
+        }
+
+        if (StringUtil.isNotNullAndNotEmpty(authenticationMethod)) {
+            //return this.client_authn_method[authn_method]().construct(
+            //      cis, cli_info, request_args, http_args, **kwargs);
+        } else {
+            return httpArgs;
+        }
+    }
+
+    public Map<String, Object> requestInfo(ClientInfo clientInfo, String method, Map<String, Object> requestArgs,
+                                           String bodyType, String authenticationMethod, boolean lax, Map<String, String> args) throws NoSuchFieldException, IllegalAccessException, MissingEndpoint, UnsupportedEncodingException, UnsupportedType {
+        if (method == null) {
+            method = this.httpMethod;
+        }
+
+        if (requestArgs == null) {
+            requestArgs = new HashMap<>();
+        }
+
+        Map<String, String> hMap = new HashMap<>();
+        for (String key : args.keySet()) {
+            if (args.get(key) != null && !specialArgs.contains(key)) {
+                hMap.put(key, args.get(key));
+            }
+        }
+
+        Message cis = this.constructMessage(clientInfo, requestArgs, args);
+
+        if (this.events != null) {
+            this.events.store("Protocol request", cis);
+        }
+
+        cis.setLax(lax);
+        Map<String, String> hArg = new HashMap<>();
+        if (StringUtil.isNotNullAndNotEmpty(authenticationMethod)) {
+            hArg = this.initAuthenticationMethod(cis, clientInfo, authenticationMethod, requestArgs, args);
+        }
+
+        if (hArg != null) {
+            if (args.containsKey("headers")) {
+                args.get("headers").update(hArg.get("headers"));
+            } else {
+                args.put("headers", hArg.get("headers"));
+            }
+        }
+
+        if (bodyType.equals("json")) {
+            args.put("contentType", "application/json");
+        }
+
+        return this.uriAndBody(cis, method, args);
+    }
+
+    public Map<String, Object> updateHttpArgs(Map<String, String> httpArgs, Map<String, Object> info) {
+        Map<String, String> hArgs = info.get("hArgs");
+        if (hArgs == null) {
+            hArgs = new HashMap<>();
+        }
+
+        if (httpArgs == null) {
+            httpArgs = hArgs;
+        } else {
+            httpArgs.update(info.get("h_args"));
+        }
+
+        info.put("httpArgs", httpArgs);
+        return info;
+    }
+
+    public Map<String, Object> doRequestInit(ClientInfo clientInfo, String bodyType, String method, String authenticationMethod,
+                                             Map<String, Object> requestArgs, Map<String, String> httpArgs, Map<String, String> args) throws NoSuchFieldException, IllegalAccessException, MissingEndpoint, UnsupportedEncodingException, UnsupportedType {
+        if (!StringUtil.isNotNullAndNotEmpty(method)) {
```

Comment:
I find this confusing as I'm used to the android text utils that have a method named the opposite: https://developer.android.com/reference/android/text/TextUtils.html#isEmpty(java.lang.CharSequence)

I think is more natural, having that the String class also has the `isEmpty` method, to name this also `isEmpty` and correct the usage to negate the condition. If not, this reads like "Is NOT a (is Not Null and Not Empty)". As long as you javadoc the methods there's no need to put the whole thing the method does in the method's name :)